src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3639,18 +3639,61 @@ void addToStringMethod(MethodSymbol printMethod)
                     refCustomModifiers: ImmutableArray<CustomModifier>.Empty,
                     explicitInterfaceImplementations: ImmutableArray<MethodSymbol>.Empty);
 
-                if (!memberSignatures.TryGetValue(targetMethod, out Symbol? existingToStringMethod))
+                var baseToStringMethod = getBaseToStringMethod();
+                bool isBaseToStringSealed = baseToStringMethod is { IsSealed: true };
+
+                if (isBaseToStringSealed)
```

Comment:
>isBaseToStringSealed [](start = 20, length = 20)

Since this local is used only in one place, consider inlining ```baseToStringMethod is { IsSealed: true }``` condition here, then we wouldn't need to supress nullable warnings when ```baseToStringMethod``` is dereferenced.

Reply:
Done